File: Strategy/Java/src/duck/MallardDuck.java
package eu.svetlin.design.patterns.strategy.duck;

import eu.svetlin.design.patterns.strategy.fly.FlyWithWings;
import eu.svetlin.design.patterns.strategy.quack.Quack;

public class MallardDuck extends Duck {
    public MallardDuck() {
        super();
        quackBehavior = new Quack();
        flyBehavior = new FlyWithWings();
    }

    @Override
    public void display() {
        System.out.println("I'm a real Mallard Duck");
    }
}

File: Strategy/TypeScript/src/duck/MallardDuck.ts
import Duck from './Duck';
import Quack from '../quack/Quack';
import FlyWithWings from '../fly/FlyWithWings';

class MallardDuck extends Duck {
  constructor() {
    super();
    this.flyBehavior = new FlyWithWings();
    this.quackBehavior = new Quack();
  }

  display() {
    console.log("I'm a real Mallard Duck");
  }
}

export default MallardDuck;
